docs/fitting.rst

```diff
@@ -15,6 +15,177 @@ initial guesses.  This model is then actually fit to the spectrum's ``flux``,
+
+   >>> # Define the spectrum
+   >>> np.random.seed(42)
+   >>> g1 = models.Gaussian1D(1, 4.6, 0.2)
+   >>> g2 = models.Gaussian1D(2.5, 5.5, 0.1)
+   >>> g3 = models.Gaussian1D(-1.7, 8.2, 0.1)
+   >>> x = np.linspace(0, 10, 200)
+   >>> y = g1(x) + g2(x) + g3(x) + np.random.normal(0., 0.2, x.shape)
+   >>> spectrum = Spectrum1D(flux=y*u.Jy, spectral_axis=x*u.um)
+
+   >>> # Define a noise region for adding the uncertainty
+   >>> noise_region = SpectralRegion(0*u.um, 3*u.um)
+   >>> spectrum = noise_region_uncertainty(spectrum, noise_region)
+
+   >>> # Derivative method
+   >>> lines = find_lines_threshold(spectrum, noise_factor=3)
+
+   >>> emission_lines = lines[lines['line_type'] == 'emission'] #doctest:+SKIP
+         line_center    line_type line_center_index
+              um                                   
+      ----------------- --------- -----------------
+      4.572864321608041  emission                91
+      4.824120603015076  emission                96
+      5.477386934673367  emission               109
+       8.99497487437186  emission               179
+
+   >>> absorption_lines = lines[lines['line_type'] == 'absorption'] #doctest:+SKIP
+         line_center    line_type  line_center_index
+              um
+      ----------------- ---------- -----------------
+      8.190954773869347 absorption               163
+
+An example using the `~specutils.fitting.find_lines_derivative`:
+
+.. code-block:: python
+
+   >>> import numpy as np
+   >>> from astropy.modeling import models
+   >>> import astropy.units as u
+   >>> from specutils import Spectrum1D, SpectralRegion
+   >>> from specutils.manipulation import noise_region_uncertainty
+   >>> from specutils.fitting import find_lines_derivative
+
+   >>> # Define the spectrum
+   >>> np.random.seed(42)
+   >>> g1 = models.Gaussian1D(1, 4.6, 0.2)
+   >>> g2 = models.Gaussian1D(2.5, 5.5, 0.1)
+   >>> g3 = models.Gaussian1D(-1.7, 8.2, 0.1)
+   >>> x = np.linspace(0, 10, 200)
+   >>> y = g1(x) + g2(x) + g3(x) + np.random.normal(0., 0.2, x.shape)
+   >>> spectrum = Spectrum1D(flux=y*u.Jy, spectral_axis=x*u.um)
+
+   >>> # Define a noise region for adding the uncertainty
+   >>> noise_region = SpectralRegion(0*u.um, 3*u.um)
+
+   >>> # Derivative method
+   >>> lines = find_lines_derivative(spectrum, flux_threshold=0.75)
+
+   >>> emission_lines = lines[lines['line_type'] == 'emission'] #doctest:+SKIP
+         line_center    line_type line_center_index
+              um                                   
+      ----------------- --------- -----------------
+      4.522613065326634  emission                90
+      5.477386934673367  emission               109
+
+   >>> absorption_lines = lines[lines['line_type'] == 'absorption'] #doctest:+SKIP
+         line_center    line_type  line_center_index
+              um                                    
+      ----------------- ---------- -----------------
+      8.190954773869347 absorption               163
+
+
```

Comment:
```suggestion
While it might be surprising that these tables do not contain more information
about the lines, this is because the "toolbox" philosophy of `specutils` aims to
keep such functionality in separate distinct functions.  See :doc:`analysis` for
functions that can be used to fill out common line measurements more
completely.

```

Reply:
will add